logger/logger.go

```diff
@@ -318,8 +317,12 @@ func (l *Logger) CounterD(title string, value int, data map[string]interface{})
 		// Use a Int64UpDownCounter counter instead of a Int64Counter DataDog chose not to follow the OTEL spec
 		// this causes the first add to not be counted and reported
 		// https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3654
-		counter := metric.Must(meter).NewInt64UpDownCounter(title)
-		counter.Add(context.Background(), int64(value), getLabels(data)...)
+		counter, err := meter.AsyncInt64().UpDownCounter(title)
```

Comment:
Where do async errors here end up?

Reply:
They end up on the normal kayvee log path. (usually DD)

Reply:
They would follow the normal kv log path.  (Usually datadog)